cmd/image-builder/build.go

```diff
@@ -38,16 +41,32 @@ func buildImage(pbar progress.ProgressBar, res *imagefilter.Result, osbuildManif
 	}
 	if opts.WriteBuildlog {
 		if err := os.MkdirAll(opts.OutputDir, 0755); err != nil {
-			return fmt.Errorf("cannot create buildlog base directory: %w", err)
+			return "", fmt.Errorf("cannot create buildlog base directory: %w", err)
 		}
-		p := filepath.Join(opts.OutputDir, fmt.Sprintf("%s.buildlog", outputNameFor(res)))
+		p := filepath.Join(opts.OutputDir, fmt.Sprintf("%s.buildlog", basename))
 		f, err := os.Create(p)
 		if err != nil {
-			return fmt.Errorf("cannot create buildlog: %w", err)
+			return "", fmt.Errorf("cannot create buildlog: %w", err)
 		}
 		defer f.Close()
 
 		osbuildOpts.BuildLog = f
 	}
-	return progress.RunOSBuild(pbar, osbuildManifest, res.ImgType.Exports(), osbuildOpts)
+	if err := progress.RunOSBuild(pbar, osbuildManifest, res.ImgType.Exports(), osbuildOpts); err != nil {
+		return "", err
+	}
+	// Rename *sigh*, see https://github.com/osbuild/images/pull/1039
+	// for my preferred way. Every frontend to images has to duplicate
+	// similar code like this.
+	pipelineDir := filepath.Join(opts.OutputDir, res.ImgType.Exports()[0])
+	srcName := filepath.Join(pipelineDir, res.ImgType.Filename())
+	imgExt := strings.SplitN(res.ImgType.Filename(), ".", 2)[1]
+	dstName := filepath.Join(opts.OutputDir, fmt.Sprintf("%s.%v", basename, imgExt))
+	if err := os.Rename(srcName, dstName); err != nil {
+		return "", fmt.Errorf("cannot rename artifact to final name: %w", err)
+	}
+	// best effort, remove the now empty pipeline export dir from osbuild
+	_ = os.Remove(pipelineDir)
```

Comment:
Wont `os.Rename()` fail if the source and destination are on different filesystems?  Wouldn't it be bad if it fails and also deletes the source?  Could instead let the user know that the move failed but the artifact is available at `pipelineDir`.

Reply:
> Wont `os.Rename()` fail if the source and destination are on different filesystems?

To be clear, I bring this up because I think it might be a very common use case.

Reply:
Yes, os.Rename() will fail if src and destination are on different file-systems. This (optimistically) assumes that the "export" directory under outputDir is on the same filesystem and it will result in confusing errors if not. The only error condition I can think of is if someone has symlinks in $outputDir that matches the name of the exported pipeline, e.g. $outputDir/ec2 -> /mnt/other-disk" (which is hopefully rare but https://xkcd.com/1172/)

I can look into fixing this in osbuild to have an option there to not hardcode the output dir to the export filename or alternatively instead of exporting into a subdir prefix the pipeline-name if the concern is clashes.

Reply:
ah, I think I misunderstood the meaning of `pipelineDir`.